### PR TITLE
Add staff sling and ammo

### PR DESCRIPTION
**Pull request by snipercup**
#### Summary
```SUMMARY: Content "Add staff sling and ammo"```

#### Purpose of change
To add staff sling and ammunition

#### Describe the solution
While wielding the staff sling, any rocks in the Characters inventory or nearby will automatically be loaded and fired at the closest target. When ammunition is expended the weapon can be used as a quarter staff. There is a very decent chance of recovering rocks allowing multiple uses of limited ammunition.

#### Describe alternatives you've considered
N/A

#### Testing
Use the weapon and ammunition during gameplay

#### Additional context
This content was made by @TjKing293, I'm just doing the PR.

### CODE REVIEW

**Comment by dissociativity** [comment]
Is there a reason that this does not use pebbles like the other sling?

**Comment by TJKing293** [comment]
it was primarily made to throw small pots of burning oil over the walls of cities. It could in theory throw pebbles but it's range is close to a Sheppard sling just with the ability to throw heavier objects.

**Comment by dissociativity** [comment]
in this case you may want to classify items like pool balls as ammunition for it too
Do you intend on making a recipe for it and perhaps have it spawn in museums and such like other primitive tools?

**Comment by TJKing293** [comment]
I apologize. The recipe was a bit of a oversight. Will be adding it soon. And yes a pool ball would make an excellent projectile.

File: data/json/items/ammo.json [hunk 1]
```diff
@@ -19,6 +19,18 @@
     "//2": "1 battery is 1 kJ of energy. 40 batteries is an alkaline C-cell.",
     "fuel": { "energy": 1 }
   },
+  {
+    "type": "AMMO",
+    "id": "rock",
```
Comment (Rail-Runner) [inline data/json/items/ammo.json]: Why are you adding second "rock" item? There is already one existing, and I doubt the staff could be throwing 6L-volume rocks.

File: data/json/items/ammo.json [hunk 2]
```diff
@@ -19,6 +19,18 @@
...
+    "copy-from": "rock",
+    "name": "rock",
+    "description": "An ordinary rock",
```
Comment (Night-Pryanik) [inline data/json/items/ammo.json]: Period in the end.

File: data/json/items/ammo.json (hunk 1 above)
Comment (TJKing293) [reply data/json/items/ammo.json]: going to be changing that

File: data/json/items/ammo.json (hunk 2 above)
Comment (TJKing293) [reply data/json/items/ammo.json]: thank you

Comment (snipercup) [inline data/json/items/ammo.json]:
```suggestion
    "description": "An ordinary rock.",
```

Comment (TJKing293) [reply data/json/items/ammo.json]: change made. finished making recipe and rebalancing staff sling. Also made changes to ammo weight and volume.

**Comment by dissociativity** [comment]
> 
> in its current state it is literally a quarterstaff with a standard sling on the end. I'm not sure how to logically justify nerfing it.

if it's literally a quarterstaff with with a sling attached, why does it have bashing 25 where again, a quarterstaff has far less of 19?

The sling on the end doesn't really make it any better at bashing lol.

**Comment by TJKing293** [comment]
> 
> 
> > in its current state it is literally a quarterstaff with a standard sling on the end. I'm not sure how to logically justify nerfing it.
> 
> if it's literally a quarterstaff with with a sling attached, why does it have bashing 25 where again, a quarterstaff has far less of 19?
> 
> The sling on the end doesn't really make it any better at bashing lol.

it was changed to 19 from 25. I changed the recipe to use the long stick instead of the quarterstaffand added pool balls in addition to rocks for ammo.

File: data/json/items/ammo.json (hunk 1 above)
Comment (kevingranade) [reply data/json/items/ammo.json]: This has been the holdup for slings for a very long time, making seperate versions of mundane items to transform them into sling ammunition is not ok.

**Comment by TJKing293** [comment]
well I can do one of three things

   1. modify the rock item to have ranged damage and some effects along with the characteristics
   2.Make a "sling rock" ammo
   3.say that pebbles when used with this weapon go faster therefore harder and further

**Comment by kevingranade** [comment]
I prefer the third option, treat slinging as "tool assisted throwing".

**Comment by snipercup** [comment]
@TJKing293 changed some things. It does not show any errors anymore and there's no duplicate items. THe rock still works as before but it is now also ammo. This is the result:

image


image


image


image

File: data/json/items/ammo.json (hunk 1 above)
Comment (snipercup) [reply data/json/items/ammo.json]: There is now only one rock item, no seperate version

**Comment by Orinyau** [comment]
Is "capacity" the size of it's "magazine"? 

If so, I don't think it should be possible to use this as a melee weapon, while it is loaded.

**Comment by snipercup** [comment]
> 
> 
> Is "capacity" the size of it's "magazine"?
> 
> If so, I don't think it should be possible to use this as a melee weapon, while it is loaded.

Thanks for your question.
Actually the staff can't be loaded and used at the same time. Just like the slingshot, when you press f to aim it gets loaded and you can fire. When you press esc to cancel, the ammo is unloaded:
image


When it is unloaded you can use it as a melee weapon:
image

**Comment by snipercup** [comment]
Added recipe:
image

**Comment by Orinyau** [comment]
Okay, thanks for clarifying. Sounds good to me.

File: data/json/items/ranged.json [hunk 3]
```diff
@@ -57,6 +57,40 @@
...
+      "DURABLE_MELEE",
+      "NONCONDUCTIVE",
+      "SHEATH_SPEAR",
+      "ALWAYS_TWOHAND"
+    ],
+    "techniques": [ "WBLOCK_2", "RAPID", "SWEEP" ],
+    "skill": "throw",
+    "ammo": "rock",
+    "weight": "2000 g",
+    "volume": "6 L",
```
Comment (TJKing293) [inline data/json/items/ranged.json]:
```suggestion
    "volume": "3 L",
```

File: data/json/recipes/weapon/ranged.json [hunk 4]
```diff
@@ -14,6 +14,18 @@
...
+  {
+    "type": "recipe",
+    "result": "staff_sling",
+    "category": "CC_WEAPON",
+    "subcategory": "CSC_WEAPON_RANGED",
+    "skill_used": "fabrication",
+    "difficulty": 3,
+    "time": "1 h",
+    "autolearn": true,
+    "qualities": [ { "id": "HAMMER", "level": 3 } ],
```
Comment (TJKing293) [inline data/json/recipes/weapon/ranged.json]:
```suggestion
    "qualities": [ { "id": "HAMMER", "level": 3 }, { "id": "SAW_W", "level":  1 } ],
```

File: data/json/recipes/weapon/ranged.json [hunk 5]
```diff
@@ -14,6 +14,18 @@
...
+    "components": [ [ [ "stick_long", 1 ] ], [ [ "sling", 1 ] ] ]
```
Comment (TJKing293) [inline data/json/recipes/weapon/ranged.json]:
```suggestion
    "components": [ [ [ "stick_long", 1 ] ], [ [ "sling", 1 ] ] , [ [ "nail", 1 ] ] ]
```

**Comment by TJKing293** [comment]
Final recipe
staff sling recipe

File: data/json/recipes/weapon/ranged.json [hunk 6]
```diff
@@ -14,6 +14,18 @@
     "using": [ [ "welding_standard", 40 ] ],
     "components": [ [ [ "pipe", 2 ] ], [ [ "hose", 2 ] ], [ [ "steel_lump", 1 ], [ "steel_chunk", 4 ], [ "scrap", 12 ] ] ]
   },
+  {
+    "type": "recipe",
+    "result": "staff_sling",
+    "category": "CC_WEAPON",
+    "subcategory": "CSC_WEAPON_RANGED",
+    "skill_used": "fabrication",
+    "difficulty": 3,
```
Comment (TJKing293) [inline data/json/recipes/weapon/ranged.json]:
```suggestion
    "difficulty": 3,
    "skills_required": [ "throw", 1 ],
```

Comment (TJKing293) [inline data/json/recipes/weapon/ranged.json]: "difficulty": 3,
 "skills_required": [ "throw", 1 ],

Comment (snipercup) [inline data/json/recipes/weapon/ranged.json]:
```suggestion
    "difficulty": 3,
    "skills_required": [ "throw", 1 ],
```

Comment (snipercup) [reply data/json/recipes/weapon/ranged.json]: inserted

**Comment by TJKing293** [comment]
as far as I'm concerned when the checks are done it is ready to merge. unless somebody else has any objections.

**Comment by RDru** [comment]
Getting this error on load:
```

 DEBUG    : warnings for type staff_sling:
there is no actual ammo of type rock defined


 FUNCTION : void Item_factory::check_definitions() const
 FILE     : src/item_factory.cpp
 LINE     : 1162

```

**Comment by snipercup** [comment]
I just downloaded the latest from https://cataclysmdda.org/experimental/. Cannot reproduce your issue.

**Comment by RDru** [comment]
ok, found override for rock in blazemod and that is causing this error

**Comment by snipercup** [comment]
Ah, that explains. Where is Blazemod hosted? Maybe I can do something to fix it?

**Comment by RDru** [comment]
here on this git repo in mods :)

**Comment by snipercup** [comment]
Okay I can fix that.

**Pull request merged**